multicluster/cmd/multicluster-controller/leader.go

```diff
@@ -123,11 +123,13 @@ func runLeader(o *Options) error {
 	if err = staleController.SetupWithManager(mgr); err != nil {
 		return fmt.Errorf("error creating StaleResCleanupController: %v", err)
 	}
-	go staleController.Run(stopCh)
 
 	klog.InfoS("Leader MC Controller Starting Manager")
 	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
 		return fmt.Errorf("error running Manager: %v", err)
 	}
+
+	go staleController.Run(stopCh)
```

Comment:
Hi, I don't think this change will fix the issue, please check my comment on the issue https://github.com/antrea-io/antrea/issues/6152#issuecomment-2327962427.
Even we started this after the Controller Manager, it doesn't guarantee that caches are ready before it.

Reply:
Ok , i will make the required changes in the code and raise another pr.